### ISSUE TITLE
Immer.js interop problem

### ISSUE THREAD

**Issue by ArturBaybulatov** [sylvainpolletvillard/ObjectModel#123]
When objectmodel used in conjunction with immer, the following error comes up:

```js
import produce from 'immer';
import {ArrayModel} from 'objectmodel';


produce(['a', 'b'], arr => { arr.push('c') })
// OK: ["a", "b", "c"]

produce(ArrayModel(String)(['a', 'b']), arr => { arr.push('c') })
// Custom Formatter Failed: Uncaught TypeError: 'getPrototypeOf' on proxy:
//   proxy target is non-extensible but the trap did not return its actual prototype
```

**Comment by sylvainpolletvillard** [comment]
Hello,

What version of immer are you using ? I can't reproduce the exception with the latest version, but the return value of produce() is `null`: https://codesandbox.io/s/amazing-feistel-w8yg1

I'm not familiar with immer but it looks like a bug. I just don't know what operation immer is doing on the object to get to this result

**Comment by ArturBaybulatov** [comment]
Latest versions of both: objectmodel 4.0.6, immer 6.0.1. The error is visible, when the code is run in a browser console. Although, looks like it's only a warning, since it doesn't stop the execution.

![image](https://user-images.githubusercontent.com/3109399/76738932-90226400-677c-11ea-97ff-2e17f68368bb.png)

Actually, I've created this issue, because I had a different error in my project code, which is relatied to objectmodel and immer. The original error was the following:

![image](https://user-images.githubusercontent.com/3109399/76742041-8d763d80-6781-11ea-90b1-12f4fab6abb7.png)

My code was the following:

![image](https://user-images.githubusercontent.com/3109399/76741892-59028180-6781-11ea-97d8-0b1bdcf06693.png)

For now, I've worked around the problem, configuring immer to disable auto freezing:

```js
import {setAutoFreeze} from 'immer';

setAutoFreeze(false);
```

I'm gonna close the issue, since the original problem turned out to be just a warning, and not a real error. Thanks!
